Source/TrainProject/Program.cs

```diff
@@ -0,0 +1,194 @@
+        public static List<Passenger> passengerList { get; set; }
+        public static TimeSpan timer;
+        static readonly object _lock = new object();
+
+        public static void Main(string[] args)
+        {
+            CreateDatabase();
+
+            var trainPlaner1 = new TrainPlaner(trainList, 2).FollowSchedule(scheduleList).AddPassengers(passengerList).CrossoverControll();
+            var trainPlaner2 = new TrainPlaner(trainList, 3).FollowSchedule(scheduleList).AddPassengers(passengerList).CrossoverControll();
+
+            CreateTrainPlaner testTrainPlaner1 = new CreateTrainPlaner(trainPlaner1);
+            CreateTrainPlaner testTrainPlaner2 = new CreateTrainPlaner(trainPlaner2);
+
+            StartTimer(testTrainPlaner1, testTrainPlaner2);
+        }
+
+        public static void ArrivingEndStation(CreateTrainPlaner train)
+        {
+            bool check = false;
+            string station;
+            while(check == false)
+            {
+                if(TimeSpan.Parse(train.trainSchedules[2].arrivalTime).Subtract(TimeSpan.FromMinutes(3)) == timer)
+                {
+                    if(train.trainSchedules[2].stationId == stationList[2].id)
+                    {
+                        Console.WriteLine($"{train.train.name} will arrive to {stationList[2].stationName} in 3 minutes, setting switch to left");
+                        while(check == false)
+                        {
+                            if(train.trainSchedules[2].arrivalTime == timer.ToString())
+                            {
+                                station = stationList.Where(p => p.id == train.trainSchedules[2].stationId).ToList().Select(p => p.stationName).First();
+                                Console.WriteLine($"{train.train.name} arrived to its end station {station} and {train.passengers.Count} passenger(s) got off the train");
+                                check = true;
+                            }
+                        }
+                    }
+                    if (train.trainSchedules[2].stationId == stationList[0].id)
+                    {
+                        Console.WriteLine($"{train.train.name} will arrive to {stationList[0].stationName} in 3 minutes, setting switch to left");
+                        while (check == false)
+                        {
+                            if (train.trainSchedules[2].arrivalTime == timer.ToString())
+                            {
+                                station = stationList.Where(p => p.id == train.trainSchedules[2].stationId).ToList().Select(p => p.stationName).First();
+                                Console.WriteLine($"{train.train.name} arrived to its end station {station} and {train.passengers.Count} passenger(s) got off the train");
+                                check = true;
+                            }
+                        }
+                    }
+                }
+            }
+        }
```

Comment:
This Method seems a bit big and nestled. Could possibly be broken up into parts? :)

Reply:
Agree